apps/frontend/src/app/(app)/space/reading-list/page.tsx

```diff
@@ -1,5 +1,218 @@
+"use client"
+
+import React, { useEffect, useState, useRef } from "react"
+import { useAuth } from "@/src/contexts/AuthContext"
+import { useSpace } from "@/src/hooks/useSpace"
+import { Icon } from "@iconify-icon/react"
+import { PlusIcon } from "@radix-ui/react-icons"
+import useReadingStore from "@/src/lib/store/reading.store"
+import { type ReadingItem } from "@/src/lib/@types/Items/Reading"
+
 const ReadingListPage: React.FC = () => {
-  return <div className="text-gray-color p-16">reading list page</div>
+  const { session } = useAuth()
+  const { spaces } = useSpace() || { spaces: [] }
+  const {
+    readingItems,
+    fetchReadingList,
+    addItem: addItemToStore,
+    deleteItem: deleteItemFromStore,
+  } = useReadingStore()
+
+  const [loading, setLoading] = useState(true)
+  const [blockId, setBlockId] = useState<string | null>(null)
+
+  useEffect(() => {
+    const loadReadingList = async () => {
+      setLoading(true)
+      const readingListSpace = spaces.find(
+        (space) => space.name.toLowerCase() === "reading list"
+      )
+      if (readingListSpace) {
+        const fetchedBlockId = readingListSpace.blocks[0]
+        setBlockId(fetchedBlockId)
+        await fetchReadingList(session, fetchedBlockId)
+      }
+      setLoading(false)
+    }
+
+    if (spaces.length > 0 && !blockId) {
+      loadReadingList()
+    }
+  }, [spaces, session, fetchReadingList, blockId])
+
+  const [addingItem, setAddingItem] = useState(false)
+  const [newItemTitle, setNewItemTitle] = useState("")
+  const [newItemUrl, setNewItemUrl] = useState("")
+  const [newItemDescription, setNewItemDescription] = useState("")
+  const textareaRefTitle = useRef<HTMLTextAreaElement>(null)
+  const textareaRefUrl = useRef<HTMLTextAreaElement>(null)
+  const textareaRefDescription = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
+    const textarea = textareaRefTitle.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemTitle])
+
+  useEffect(() => {
+    const textarea = textareaRefUrl.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemUrl])
+
+  useEffect(() => {
+    const textarea = textareaRefDescription.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemDescription])
```

Comment:
Having too many useEffects can cause un-necessary re-renders and other issues.

Try combining these